src/communication.ts

```diff
@@ -1,5 +1,8 @@
 import { io, Socket } from 'socket.io-client';
 import { getSettings } from './settings';
+import { Message } from './util/message';
+
+type Subscriber = (message: string) => void;
```

Comment:
what if this needs to be async?

Reply:
this should return a Message object, that would clean a lot of code up which is always now needing to convert this to a Message object. New Task

Reply:
need to discuss this as i have some concerns